lib/kubeclient/common.rb

```diff
@@ -78,7 +80,7 @@ def initialize_client(
       @api_version = version
       @headers = {}
       @ssl_options = ssl_options
-      @auth_options = auth_options
+      @auth_options = Marshal.load(Marshal.dump(auth_options))
```

Comment:
I presume the goal here is deep copy?

I'm not very familiar with `Marshal`, trying to understand whether this is always safe...
A particular scenario to consider: auth_options will frequently come from a parsed kubeconfig e.g.:
```
config = Kubeclient::Config.read(ENV['KUBECONFIG'] || '/path/to/.kube/config')
context = config.context
Kubeclient::Client.new(
  context.api_endpoint,
  'v1',
  ssl_options: context.ssl_options,
  auth_options: context.auth_options  # <-- under external control?
)
```
[Marshal docs warn against de-serializing untrusted input](https://ruby-doc.org/core-3.1.2/Marshal.html#module-Marshal-label-Security+considerations).  But here we don't quite have untrusted input. The kubeconfig can be parsed into "safe_load" types — arbitrary mix of hashes, arrays, strings, numbers, booleans and dates -> Marshal.dump -> Marshal.load.
Probably it can only deserialize into same "safe" types?

Reply:
I googled more and this appears to be [the idiomatic way](https://stackoverflow.com/a/17913436/239657) to deep copy using ruby stdlib.  :heavy_check_mark: 

The other "obvious" contender is ActiveSupport [.deep_dup](https://guides.rubyonrails.org/active_support_core_extensions.html#deep-dup) but kubeclient being a rails-independent library probably shouldn't pull in ActiveSupport's monkey patches.

Finally, a shallow copy might suffice here?

Reply:
I was following the [the idiomatic way](https://stackoverflow.com/a/17913436/239657) only. I can switch it to shallow copy